### PR TITLE
Bug fixes for duplicate copies and scope name

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
- Prevent adding duplicates in `_tracked` data structure.
- Change the scope name to have the .scope suffix  …
- nit fixes; correcting the systemd filename and other unit test fixes.

---

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1548)

### CODE REVIEW

File: azurelinuxagent/common/cgroupstelemetry.py [hunk 1]
```diff
@@ -51,18 +51,19 @@ def track_cgroup(cgroup):
...
         with CGroupsTelemetry._rlock:
-            CGroupsTelemetry._tracked.append(cgroup)
-            logger.info("Started tracking new cgroup: {0}, path: {1}".format(cgroup.name, cgroup.path))
+            if not CGroupsTelemetry.is_tracked(cgroup.name, cgroup.controller, cgroup.path):
+                CGroupsTelemetry._tracked.append(cgroup)
+                logger.info("Started tracking new cgroup: {0}, path: {1}".format(cgroup.name, cgroup.path))
 
     @staticmethod
-    def is_tracked(name, controller):
+    def is_tracked(name, controller, path):
```
Comment (narrieta) [inline azurelinuxagent/common/cgroupstelemetry.py]: why do we need the path on top of the name and controller?

Comment (vrdmr) [reply azurelinuxagent/common/cgroupstelemetry.py]: Discussed offline. Will change this.

File: tests/common/test_cgroupstelemetry.py [hunk 2]
```diff
@@ -429,6 +435,9 @@ def test_telemetry_with_tracked_cgroup(self):
             with patch("azurelinuxagent.ga.exthandlers.ExtHandlerInstance.get_log_dir", lambda *_: self.log_dir) as patch_get_log_dir:
                 self.ext_handler_instance.launch_command(command)
 
+        self.assertTrue(CGroupsTelemetry.is_tracked("/sys/fs/cgroup/cpu/walinuxagent.extensions/foobar_1.0.0"))
```
Comment (pgombar) [inline tests/common/test_cgroupstelemetry.py]: Maybe use the same path building as in `test_cgroupconfigurator` class, e.g.:
`self.assertTrue(CGroupsTelemetry.is_tracked(os.path.join(self.cgroups_file_system_root, "cpu", "walinuxagent.extensions/Microsoft.Compute.TestExtension_1.2.3")))`

Comment (vrdmr) [reply tests/common/test_cgroupstelemetry.py]: Discussed offline. Added a new TODO for systemd test as well.

**Pull request merged**